Sources/ScipioKit/Producer/Cache/CacheSystem.swift

```diff
@@ -306,3 +312,32 @@ public struct VersionFileDecoder {
         )
     }
 }
+
+#if compiler(>=6.0)
+
+fileprivate extension ResolvedPackage {
+
+    var pin: PinsStore.Pin? {
```

Comment:
I feel `pin` is too generic a property name, even if it's a fileprivate extension.

This property generates a pin it is using `GitRepository`

How about it be `makePinByRevision()` or something?

Reply:
Agree, I like using an optional `makePinFromRevision()` method for that. I just slightly changed the preposition.